doc/flakes/design.md

```diff
@@ -329,213 +301,154 @@ of the standard result attributes:
   we need to avoid a situation where `nixos-rebuild` needs to fetch
   its own `nixpkgs` just to do `evalModules`.)
 
-* `devShell`: A specification of a development environment in some TBD
-  format.
-
-The function argument `flakes` is an attrset that contains an
-attribute for each dependency specified in `inputs`. (Should it
-contain transitive dependencies? Probably not.) Each attribute is an
-attrset containing the `outputs` of the dependency, in addition to
-the following attributes:
-
-* `path`: The path to the flake's source code. Useful when you want to
-  use non-Nix artifacts from the flake, or if you want to *store* the
-  source code of the dependency in a derivation. (For example, we
-  could store the sources of all flake dependencies in a NixOS system
-  configuration, as a generalization of
-  `system.copySystemConfiguration`.)
-
-* `meta`: An attrset containing the following:
-
-  * `description`
-
-  * `commitHash` (or `rev`?) (not for tarball flakes): The Git commit
-    hash.
+* `devShell`: A specification of a development environment in some TBD format.
 
-  * `date`: The timestamp of the most recent commit (for Git
-    repositories), or the timestamp of the most recently modified file
-    (for tarballs).
-
-  * `revCount` (for Git flakes, but not GitHub flakes): The number of
-    ancestors of the revision. Useful for generating version strings.
-
-
-## Non-flake dependencies
-
-It may be useful to pull in repositories that are not flakes
-(i.e. don't contain a `flake.nix`). This could be done in two ways:
-
-* Allow flakes not to have a `flake.nix` file, in which case it's a
-  flake with no inputs and no outputs. The downside of this
-  approach is that we can't detect accidental use of a non-flake
-  repository. (Also, we need to conjure up an identifier somehow.)
-
-* Add a flake attribute to specifiy non-flake dependencies, e.g.
-
-  > nonFlakeInputs.foobar = github:foo/bar;
+* `self`: The path to the flake's source code
```

Comment:
This is in fact the result of the flake's output which is passed into itself. So `self.outputs.foo` should work.